### ISSUE TITLE
mininode tests broken with Python 3.4.2 (Debian Jessie)

### ISSUE THREAD

**Issue by domob1812** [bitcoin/bitcoin#13745]
The recent change in #13715 broke mininode tests on systems with Python 3.4.2 (as included in Debian Jessie aka oldstable).  This is because the method [`is_closing`](https://docs.python.org/3.4/library/asyncio-protocol.html#asyncio.BaseTransport.is_closing) is only introduced in Python 3.4.4.

A workaround would be to add a `hasattr` condition that only checks `is_closing` if it is actually available - keeping the benefits of #13715 for newer systems while not breaking old systems (basically falling back to the behaviour before #13715 for them).  If that's a desired fix, I'm happy to create a PR for it.

**Comment by MarcoFalke** [comment]
I am happy with that fallback, if the tests would otherwise be impossible to run. Just noting that you probably shouldn't run anything older than python 3.4.8 and that python 3.4 is completely EOL Q1 2019: https://devguide.python.org/#status-of-python-branches

**Comment by marcoagner** [comment]
I think a simple and explicit if condition in `send_message` would be fine so it stays obvious and simple for when Python <=3.4.2 is dropped:

```
if hasattr(self._transport, "is_closing"):                              
    NetworkThread.network_event_loop.call_soon_threadsafe(lambda: self._transport and not self._transport.is_closing() and self._transport.write(tmsg))
else:
    # is_closing not implemented before Python 3.4.4                                                             
    NetworkThread.network_event_loop.call_soon_threadsafe(lambda: self._transport and self._transport.write(tmsg))
```
Unless https://github.com/bitcoin/bitcoin/issues/13579 should be solved for Python <3.4.4 too; then another workaround should be implemented.
LMK if you are doing it or would like me to do it.

**Comment by domob1812** [comment]
Sounds good to me.  I don't know if #13579 should be solved also for old systems, but will prepare a PR with the simple workaround for now.

**Comment by domob1812** [comment]
Proposed fix is in #13747, which simply disables the new `is_closing` check for systems that don't have it (as discussed above).
